clients/java/src/test/java/io/camunda/client/adhocsubprocess/AdHocSubprocessActivitySearchTest.java

```diff
@@ -0,0 +1,173 @@
+              r.setAdHocSubprocessId(AD_HOC_SUBPROCESS_ID);
+              r.setFlowNodeId("task2");
+              r.setFlowNodeName("Task #2");
+              r.setType(TypeEnum.USER_TASK);
+              r.setDocumentation("The second task in the ad-hoc subprocess");
+              r.setTenantId("<default>");
+            }));
+
+    gatewayService.onAdHocSubprocessActivitySearch(searchQueryResult);
+
+    // when
+    final AdHocSubprocessActivityResponse response =
+        client
+            .newAdHocSubprocessActivityQuery(PROCESS_DEFINITION_KEY, AD_HOC_SUBPROCESS_ID)
+            .send()
+            .join();
+
+    // then
+    final AdHocSubprocessActivitySearchQuery request =
+        gatewayService.getLastRequest(AdHocSubprocessActivitySearchQuery.class);
+    assertThat(request.getFilter().getProcessDefinitionKey())
+        .isEqualTo(PROCESS_DEFINITION_KEY.toString());
+    assertThat(request.getFilter().getAdHocSubprocessId()).isEqualTo(AD_HOC_SUBPROCESS_ID);
+
+    assertThat(response.getItems())
+        .hasSize(2)
+        .extracting(
+            AdHocSubprocessActivity::getProcessDefinitionKey,
+            AdHocSubprocessActivity::getProcessDefinitionId,
+            AdHocSubprocessActivity::getAdHocSubprocessId,
+            AdHocSubprocessActivity::getFlowNodeId,
+            AdHocSubprocessActivity::getFlowNodeName,
+            AdHocSubprocessActivity::getType,
+            AdHocSubprocessActivity::getDocumentation,
+            AdHocSubprocessActivity::getTenantId)
+        .containsExactly(
+            tuple(
+                PROCESS_DEFINITION_KEY,
+                PROCESS_DEFINITION_ID,
+                AD_HOC_SUBPROCESS_ID,
+                "task1",
+                "Task #1",
+                AdHocSubprocessActivityType.SERVICE_TASK,
+                "The first task in the ad-hoc subprocess",
+                "<default>"),
+            tuple(
+                PROCESS_DEFINITION_KEY,
+                PROCESS_DEFINITION_ID,
+                AD_HOC_SUBPROCESS_ID,
+                "task2",
+                "Task #2",
+                AdHocSubprocessActivityType.USER_TASK,
+                "The second task in the ad-hoc subprocess",
+                "<default>"));
+  }
+
+  @Test
+  void shouldMapUnknownEnumValueToFallbackValueWithoutBreaking() {
+    // given
+    final String responseJson =
+        "{\n"
+            + "  \"items\": [\n"
+            + "    {\n"
+            + "      \"processDefinitionKey\": \"2251799813685281\",\n"
+            + "      \"processDefinitionId\": \"TestParentAdHocSubprocess\",\n"
+            + "      \"adHocSubprocessId\": \"TestAdHocSubprocess\",\n"
+            + "      \"flowNodeId\": \"unknownTask\",\n"
+            + "      \"flowNodeName\": \"Unknown Task\",\n"
+            + "      \"type\": \"START_EVENT\",\n"
+            + "      \"tenantId\": \"<default>\"\n"
+            + "    }\n"
+            + "  ]\n"
+            + "}";
```

Comment:
🙃 You can consider using text blocks to make this a bit nicer, see [this test](https://github.com/camunda/camunda/blob/main/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/TopologyControllerTest.java#L43) for an example.

Reply:
I did so on the other modules (e.g. the REST controller tests), but unfortunately this module is still Java 8.

Reply:
Ugh, yeah, the client, I forgot 😄 Nevermind, then 😅